data_steward/cdr_cleaner/cleaning_rules/remove_invalid_procedure_source_records.py

```diff
@@ -0,0 +1,74 @@
+"""
+Background
+
+Some values for procedure_source_concept_id in the procedure_occurrence table have been identified as CPT modifiers,
+which are supposed to be in the modifier_concept_id field.
+
+While the CPT modifiers could be moved to the correct field, this would leave these rows with
+procedure_source_concept_id=0, which has little utility. Therefore, these rows should be dropped.
+
+Cleaning rule to remove records where:
+-procedure_concept_id is not a standard concept in the procedure domain
+AND
+-procedure_source_concept_id is not in the procedure domain (they ARE allowed to be non-standard).
+
+"""
+
+# Project imports
+import constants.cdr_cleaner.clean_cdr as cdr_consts
+
+REMOVE_INVALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY = """
+DELETE
+FROM
```

Comment:
The query you have above for the `procedure_concept_id` returns 1,131,526 records for me, which I think is too high. From my understanding we want to delete records that have the domain_id = 'Procedure' and standard_concept != 'S', I also included all concept_class_ids that include CPT , which results with my query below and returns 11 records: 

`-- procedure_concept_id is not a standard concept in the procedure domain
WHERE p.procedure_concept_id IN (
  SELECT
    concept_id
  FROM
    `{DATASET_ID}.concept`
  WHERE
    domain_id = 'Procedure'
    AND TRIM(concept_class_id) IN ('Procedure', 'CPT4', 'CPT4 Modifier', 'CPT4 Hierarchy')
    AND standard_concept != 'S'
)
OR
-- procedure_source_concept_id is not in the procedure domain
p.procedure_source_concept_id IN (
 SELECT
    concept_id
  FROM
    `{DATASET_ID}.concept`
  WHERE
    TRIM(concept_class_id) = 'CPT4 Modifier'
)`

Also from my understanding (correct me if I'm wrong) we want the statement to have OR since it can either have a Procedure domain with a nonstandard concept or a concept_class_id of 'CPT4 Modifier'? Swapping out OR for AND returns zero records for me. Let me know your thoughts!